api/db-cdr/generate-cdr/make-bq-data.sh

```diff
@@ -0,0 +1,183 @@
+#!/bin/bash
+
+# This generates big query data that gets put in cloudsql
+# Counts and teh public and cdr data four cloudsql
+# note dev-up must be run to generate the schema
+# note run-local-data-migrations must be run to generate hard coded data from liquibase
+# note  the account must be authorized to perform gcloud and bq operations
+
+set -xeuo pipefail
+IFS=$'\n\t'
+
+
+# get options
+# --project=all-of-us-workbench-test *required
+
+# --cdr=cdr_version ... *optional
+
+USAGE="./generate-clousql-cdr/make-bq-data.sh --bq-project <PROJECT> --bq-dataset <DATASET> --workbench-project <PROJECT>"
+USAGE="$USAGE --account <ACCOUNT> --cdr-version=YYYYMMDD"
+while [ $# -gt 0 ]; do
+  echo "1 is $1"
+  case "$1" in
+    --account) ACCOUNT=$2; shift 2;;
+    --bq-project) BQ_PROJECT=$2; shift 2;;
+    --bq-dataset) BQ_DATASET=$2; shift 2;;
+    --workbench-project) WORKBENCH_PROJECT=$2; shift 2;;
+    --cdr-version) CDR_VERSION=$2; shift 2;;
+    -- ) shift; break ;;
+    * ) break ;;
+  esac
+done
+
+if [ -z "${ACCOUNT}" ]
+then
+  echo "Usage: $USAGE"
+  exit 1
+fi
+
+if [ -z "${BQ_PROJECT}" ]
+then
+  echo "Usage: $USAGE"
+  exit 1
+fi
+
+if [ -z "${BQ_DATASET}" ]
+then
+  echo "Usage: $USAGE"
+  exit 1
+fi
+
+if [ -z "${WORKBENCH_PROJECT}" ]
+then
+  echo "Usage: $USAGE"
+  exit 1
+fi
+
+if [ -z "${CDR_VERSION}" ]
+then
+  echo "Usage: $USAGE"
+  exit 1
+fi
+
+
+#Check cdr_version is of form YYYYMMDD
+if [[ $CDR_VERSION =~ ^[0-9]{4}(0[1-9]|1[0-2])(0[1-9]|[1-2][0-9]|3[0-1])$ ]]; then
+    echo "New CDR VERSION will be $CDR_VERSION"
+  else
+    echo "CDR Version doesn't match required format YYYYMMDD"
+    echo "Usage: $USAGE"
+    exit 1
+fi
+
+NEW_BQ_CDR_DATASET=cdr$CDR_VERSION
+schema_path=generate-cdr/bq-schemas
+
+# Check that bq_dataset exists and exit if not
+datasets=`bq --project=$BQ_PROJECT ls`
+if [ -z "$datasets" ]
+then
+  echo "$BQ_PROJECT.$BQ_DATASET does not exist. Please specify a valid project and dataset."
+  exit 1
+fi
+if [[ $datasets =~ .*$BQ_DATASET.* ]]; then
+  echo "$BQ_PROJECT.$BQ_DATASET exists. Good. Carrying on."
+else
+  echo "$BQ_PROJECT.$BQ_DATASET does not exist. Please specify a valid project and dataset."
+  exit 1
+fi
+
+
+# Make dataset for cdr cloudsql tables
+datasets=`bq --project=$WORKBENCH_PROJECT ls`
+echo $datasets
+if [[ $datasets =~ .*$NEW_BQ_CDR_DATASET.* ]]; then
+  echo "$NEW_BQ_CDR_DATASET exists"
+else
+  echo "Creating $NEW_BQ_CDR_DATASET"
+  bq --project=$WORKBENCH_PROJECT mk $NEW_BQ_CDR_DATASET
+fi
+
+# Copy tables we can that we need from cdr to our cloudsql cdr dataset
```

Comment:
Do we need to copy these? Or could we just export them from the CDR directly?

Reply:
I was just trying to be consistent with the theme of putting all data in one dataset and dumping that dataset.  I'm not sure if that makes anything much easier.   These tables are really small. 

I guess a question is once we make the csv dumps, what do we do with the dataset ? Delete it ?

Reply:
As discussed offline: we don't want to delete these datasets. We want researchers (or the public) to be able to use them.

Copying sounds fine, since access to the metadata may be different from access to the CDR itself.

Reply:
These tables are pretty safe.  But to catch any omop schema changes, and after our  discussion, and to be consistent with "defining a schema for workbench independently of omop" , I think i will make schema files for these and then just query these to get the data in them. And if that query breaks , we know we have a omop / workbench discrepancy we need to resolve.